From 3335c980271976caa085f45e6148e3c4934d75a2 Mon Sep 17 00:00:00 2001
From: Ke Jia <ke.a.jia@intel.com>
Date: Tue, 25 Aug 2015 01:57:47 -0400
Subject: [PATCH 0693/1164] CDH-42370 - HIVE-11624:Beeline-cli: support
 hive.cli.print.header in new CLI[beeline-cli
 branch](Ke Jia, reviewed by Ferdinand Xu)

Change-Id: Ia9dfa86738ac676e82d45e33622bb7cc15ef8a7f
---
 .../java/org/apache/hive/beeline/BeeLineOpts.java  |    9 ++++++++-
 .../org/apache/hive/beeline/cli/TestHiveCli.java   |    7 +++++++
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
index 2988089..d5088ff 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
@@ -411,7 +411,14 @@ public void setShowHeader(boolean showHeader) {
   }
 
   public boolean getShowHeader() {
-    return showHeader;
+    if (beeLine.isBeeLine()) {
+      return showHeader;
+    } else {
+      boolean header;
+      HiveConf conf = beeLine.getCommands().getHiveConf(true);
+      header = HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_CLI_PRINT_HEADER);
+      return header;
+    }
   }
 
   public void setHeaderInterval(int headerInterval) {
diff --git a/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java b/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
index c465eac..52507e2 100644
--- a/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
+++ b/beeline/src/test/org/apache/hive/beeline/cli/TestHiveCli.java
@@ -95,6 +95,13 @@ public void testSetPromptValue() {
   }
 
   @Test
+  public void testSetHeaderValue() {
+    verifyCMD(
+        "create database if not exists test;\ncreate table if not exists test.testTbl(a string, b string);\nset hive.cli.print.header=true;\n select * from test.testTbl;\n",
+        "testtbl.a testtbl.b", os, null, ERRNO_OK);
+  }
+
+  @Test
   public void testHelp() {
     verifyCMD(null, "usage: hive", os, new String[] { "-H" }, ERRNO_ARGS);
   }
-- 
1.7.9.5

